### PR TITLE
Bug fixes

### LINKED ISSUE #416: Fix "Duplicate Transaction" printf
It prints a byte array like this: 

`[skycoin.daemon:WARNING] Duplicate Transaction: [2 242 64 154 188 90 187 148 249 235 38 122 162 135 141 117 153 23 152 9 232 48 177 228 191 37 158 28 81 125 121 113]`

### PR DESCRIPTION
**Pull request by iketheadore**
* Fix issue #376 , which affect both the history page in wallet and explorer, the old `/explorer/address` api will only return `receive` coins transaction
* Make /wallet/newAddress api able to create multiple address in one api call, and update corresponding client code
* Fix failed test in src/visor/historydb package
* Fix  #416, 'Duplicate transaction' printf

### CODE REVIEW

File: src/visor/historydb/address_txn.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
+package historydb
+
+import (
+	"github.com/boltdb/bolt"
+	"github.com/skycoin/examples/aether/encoder"
```
Comment (gz-c) [inline src/visor/historydb/address_txn.go]: wrong import?

Comment (iketheadore) [inline src/visor/historydb/address_txn.go]: oh, yes, wrong import, will fix

File: src/visor/historydb/historydb.go [hunk 2]
```diff
@@ -56,9 +60,79 @@ func New(db *bolt.DB) (*HistoryDB, error) {
...
+// If we have a new added bucket, we need to reset to parse
+// blockchain again to get the new bucket filled.
+func (hd *HistoryDB) ResetIfNeed() error {
+	if hd.historyMeta.ParsedHeight() == 0 {
+		return nil
+	}
+
+	// if any of the following buckets are empty, need to reset
+	var reset bool
+	for {
```
Comment (gz-c) [inline src/visor/historydb/historydb.go]: use `if` or `switch`

File: src/visor/bucket/bucket.go [hunk 3]
```diff
@@ -161,6 +156,21 @@ func (b *Bucket) IsExist(k []byte) bool {
 	return exist
 }
 
+// IsEmpty check if the bucket is empty
+func (b *Bucket) IsEmpty() bool {
+	var notEmpty bool
```
Comment (gz-c) [inline src/visor/bucket/bucket.go]: `empty := true`
...
`if k != nil { empty = false }`
...
`return empty`

File: src/visor/readable.go [hunk 4]
```diff
@@ -418,7 +380,7 @@ type TransactionOutputJSON struct {
 	SourceTransaction string `json:"src_tx"`
 	Address           string `json:"address"` // Address of receiver
 	Coins             string `json:"coins"`   // Number of coins
-	Hours             uint64 `json:"hours"`   // Coin hours
+	Hours             string `json:"hours"`   // Coin hours
```
Comment (gz-c) [inline src/visor/readable.go]: will this affect saved wallets or other running clients on the network?

File: src/visor/historydb/historydb.go (hunk 2 above)
Comment (iketheadore) [inline src/visor/historydb/historydb.go]: use if or switch for what?  if without the for loop, we need to do the reset action in every checking.

File: src/visor/readable.go (hunk 4 above)
Comment (iketheadore) [inline src/visor/readable.go]: will reset this change

**Comment by iketheadore** [comment]
Should I remove the cmd/explorer?

File: src/visor/historydb/historydb.go (hunk 2 above)
Comment (gz-c) [inline src/visor/historydb/historydb.go]: how is the for loop different from `if hd.addrTxns.IsEmpty() || hd.addrUx.IsEmpty() || hd.txns.IsEmpty() || hd.outputs.IsEmpty() { return hd.reset() }`

**Comment by iketheadore** [comment]
Updated, please check again

File: cmd/explorer/explorer.go [hunk 5]
```diff
@@ -54,7 +54,7 @@ func getBlockChainMetaData(w http.ResponseWriter, r *http.Request) {
 
 func getAddress(w http.ResponseWriter, r *http.Request) {
 	address := r.URL.Query().Get("address")
-	resp, err := http.Get("http://127.0.0.1:6420/explorer/address?address=" + address)
+	resp, err := http.Get("http://127.0.0.1:6420/explorer/transactions?address=" + address)
```
Comment (gz-c) [inline cmd/explorer/explorer.go]: undo

File: cmd/explorer/server/routes/api.js [hunk 6]
```diff
@@ -95,7 +95,7 @@ router.get('/currentBalance', (req, res) => {
 
 // Get the block details
 router.get('/coinSupply', (req, res) => {
-  axios.get("http://127.0.0.1:6420/explorer/getEffectiveOutputs")
+  axios.get("http://127.0.0.1:6420/explorer/coinSupply")
```
Comment (gz-c) [inline cmd/explorer/server/routes/api.js]: undo

**Comment by iketheadore** [comment]
updated, check please.

File: src/visor/readable.go [hunk 7]
```diff
@@ -438,26 +400,25 @@ func NewTransactionOutputJSON(ux coin.TransactionOutput, srcTx cipher.SHA256) Tr
 
 	o.Address = ux.Address.String()
 	o.Coins = StrBalance(ux.Coins)
-	o.Hours = ux.Hours
+	o.Hours = strconv.FormatUint(ux.Hours, 10)
```
Comment (gz-c) [inline src/visor/readable.go]: undo api change

**Pull request merged**